src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/ExprToolCompMiningSpeed.java

```diff
@@ -0,0 +1,128 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
+import ch.njol.util.Math2;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.components.ToolComponent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Consumer;
+
+@Name("Tool Component - Mining Speed")
+@Description("The default mining speed of a tool.")
+@Examples({
+	"set the default mining speed of {_tool} to 10",
+	"",
+	"set {_component} to the tool component of {_tool}",
+	"set the mining speed of {_component} to 5",
+	"set the tool component of {_tool} to {_component}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class ExprToolCompMiningSpeed extends PropertyExpression<Object, Number> {
+
+	static {
+		Skript.registerExpression(ExprToolCompMiningSpeed.class, Number.class, ExpressionType.PROPERTY,
+			"[the] [default] mining speed of %itemstacks/itemtypes/slots/toolcomponents%");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr(exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected Number @Nullable [] get(Event event, Object[] source) {
+		return get(source, object -> {
+			if (object instanceof ToolComponent component)
+				return component.getDefaultMiningSpeed();
+
+			ItemStack itemStack = ItemUtils.asItemStack(object);
+			if (itemStack == null)
+				return null;
+			return itemStack.getItemMeta().getTool().getDefaultMiningSpeed();
+		});
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, REMOVE, ADD -> CollectionUtils.array(Number.class);
+			default -> null;
```

Comment:
Might be worth adding reset, the javadocs for the `getDefaultMiningSpeed` and `setDefaultMiningSpeed` method mentions `1.0` is the standard mining speed making me feel like that's more of a default.